tests/unit/utils/validation-core.test.ts

```
@@ -0,0 +1,228 @@
+import { describe, it, expect } from 'vitest';
+
+import {
+  isRecord,
+  isNonEmptyString,
+  validateStringField,
+  validateEnumField,
+  validateRequired,
+  validateObjectField,
+  toString,
```

Biome (1.9.4)

[error] Do not shadow the global "toString" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)